python-api/tests/integrationtests/test_controller.py

```diff
@@ -3,725 +3,343 @@
+        self.controller.on_preview_port_added(test_cb)
+
+        self.setup_sources()
+        self.run_mainloop(timeout=5)
+
+        self.log.info("test_cb called with: %s", test_cb.call_args_list)
+        test_cb.assert_any_call(3003, 1, 7)
+        test_cb.assert_any_call(3004, 1, 8)
+        assert test_cb.call_count == 2
+
+
+class TestNewRecord(IntegrationTestbase):
+    """ Test new_record method
+    """
+
+    def wait_for_sources(self, count):
+        """ Blocks until the Server has reported, that the right number of
+        preview-ports are is started
+        """
+        self.log.info("waiting for Server to start preview-port-outputs")
+        self.serv.wait_for_output('tcpserversink name=sink', count=count)
+
+    def setup_sources(self):
+        """ Starts some Test-Video streams and waits until they are ready
+        """
+        self.log.info("starting 2 test-video sources")
+        self.new_test_video()
+        self.new_test_video()
+
+        self.log.info("waiting for the test-video sources to come up")
+        self.wait_for_sources(count=2)
+
+    def test_filename_changes(self):
+        """ Tests, that the Server creates a new File each time
+        new_record is called
+        """
+        test_filename = "gst-test-{0}.data".format(
+            random.randint(0, sys.maxsize))
+
+        self.log.info("asserting recording-file are not aready existing"
+                      "(test_filename=%s)", test_filename)
+        assert not os.path.exists(test_filename)
+        assert not os.path.exists(test_filename + '.000')
+
+        self.setup_server(record_file=test_filename)
+        self.setup_controller()
+        self.setup_sources()
+
+        self.log.info("asserting server created a recording-file")
+        assert os.path.exists(test_filename)
+        assert not os.path.exists(test_filename + '.000')
+
+        self.log.info("starting a new recording")
+        assert self.controller.new_record()
+
+        self.log.info("asserting server created a new recording-file")
+        assert os.path.exists(test_filename)
+        assert os.path.exists(test_filename + '.000')
+
+        os.remove(test_filename)
+        os.remove(test_filename + '.000')
+
+    def test_record_file_grows(self):
+        """ Tests, that the Server actually writes Data into the record-file
+        """
+        test_filename = "gst-test-{0}.data".format(
+            random.randint(0, sys.maxsize))
+
+        self.log.info("asserting recording-file are not aready existing"
+                      "(test_filename=%s)", test_filename)
+        assert not os.path.exists(test_filename)
+
+        self.setup_server(record_file=test_filename)
+        self.setup_controller()
+        self.setup_sources()
+
+        self.log.info("asserting server created a recording-file")
+        assert os.path.exists(test_filename)
+
+        self.log.info("sleeping for 5 frames")
+        time.sleep(5 / 25)
```

Comment:
We could but I doubt it will be faster then just waiting 200ms -- and as long as the test is reliable enough I don't see a point in restructuring it. I'll add a sourcecode-comment, though.